src/test/java/io/reactivex/internal/operators/completable/CompletableDelayTest.java

```diff
@@ -120,4 +127,84 @@ public void errorDelayed() {
 
         to.assertFailure(TestException.class);
     }
+
+    @Test
+    public void errorDelayedSubscription() {
+        TestScheduler scheduler = new TestScheduler();
+
+        TestObserver<Void> to = Completable.error(new TestException())
+                .delaySubscription(100, TimeUnit.MILLISECONDS, scheduler)
+                .test();
+
+        to.assertEmpty();
+
+        scheduler.advanceTimeBy(90, TimeUnit.MILLISECONDS);
+
+        to.assertEmpty();
+
+        scheduler.advanceTimeBy(15, TimeUnit.MILLISECONDS);
+
+        to.assertFailure(TestException.class);
+    }
+
+    @Test
+    public void errorDelayedSubscriptionDisposeBeforeTime() {
+        TestScheduler scheduler = new TestScheduler();
+
+        Completable result = Completable.complete()
+                .delaySubscription(100, TimeUnit.MILLISECONDS, scheduler);
+        Observer<Void> o = TestHelper.mockObserver();
+        TestObserver<Void> to = new TestObserver<Void>(o);
+
+        result.subscribe(to);
+        to.assertEmpty();
+
+        scheduler.advanceTimeBy(90, TimeUnit.MILLISECONDS);
+        to.dispose();
+
+        scheduler.advanceTimeBy(15, TimeUnit.MILLISECONDS);
+
+        to.assertEmpty();
+        verify(o, never()).onComplete();
```

Comment:
No need to talk to  a mocked observer, TestObserver.assertEmpty() already verifies these.